### PR TITLE
Streamlining configure-ec2

### LINKED ISSUE #42: Migrate configure-ec2 to JS
Bash scripting is not necessarily a sustainable method of development for our project.  Ideally, we would like to consolidate all Autostack code into Node which will improve understandability of the code and make tracing code execution easier.  While the use of commands will not be entirely eliminated, it should be possible to reduce our reliance on the command line.

### PR DESCRIPTION
**Pull request by nlipiarski**
Moved instance type detection and world/flavor syncing from `configure-ec2` to JS.  Additionally, switched from executing the service as root to executing it as the ec2-user user instead which helps me sleep at night.  These changes should also make it easier to migrate `uploadworld` and `deletestack` away from bash scripts which is also on my radar.  

Closes #42

### CODE REVIEW

File: AutoStack/configure-ec2 [hunk 1]
```diff
@@ -7,46 +7,21 @@ fi
...
-	echo "Unkown instance type '$INSTANCE_TYPE'" > error.log
-	exit
-fi
-
-echo "{" > autostack-scripts/instance_data.json
-echo "\"available_ram\":\"$AVAILABLE_RAM\"," >> autostack-scripts/instance_data.json
-echo "\"world_url\":\"$WORLD_URL\"," >> autostack-scripts/instance_data.json
-echo "\"stack_name\":\"$STACK_NAME\"" >> autostack-scripts/instance_data.json
-echo "}" >> autostack-scripts/instance_data.json
+chown -R ec2-user:ec2-user *
```
Comment (carlquistb) [inline AutoStack/configure-ec2:24]: in this line, are we saying "recursively everything in the ec2-user directory should be owned by ec2-user?

File: AutoStack/Main.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+const execSync = require("child_process").execSync;
+const http = require("http");
+const MineShell = require("./MineShell.js");
```
Comment (carlquistb) [inline AutoStack/Main.js]: don't we want to take the suffix out from the require statement?

File: AutoStack/configure-ec2 (hunk 1 above)
Comment (nlipiarski) [reply AutoStack/configure-ec2:24]: Yes

**Review by carlquistb — APPROVED** [review]
as long as you are ok with the ```require("./MineShell.js");``` line, looks great!

I think you are using execSync because of there not being a good mirror for the CLI sync command in the AWS SDK for javascript. I bet we can get around that gracefully, maybe it was replaced by ```cp -recursive``` or something. I'll look into it for the future!

File: AutoStack/Main.js (hunk 2 above)
Comment (nlipiarski) [reply AutoStack/Main.js]: yes! I need to rebase to get those changes

**Pull request merged**